core/src/main/java/nu/yona/server/subscriptions/service/UserService.java

```diff
@@ -76,22 +83,25 @@ public UserDTO addUser(UserDTO userResource)
 		return UserDTO.createInstanceWithPrivateData(userEntity);
 	}
 
+	@Autowired
+	UserServiceTempEncryptionContextExecutor tempEncryptionContextExecutor;
+
+	public User addUserCreatedOnBuddyRequest(UserDTO buddyUserResource, String tempPassword)
+	{
+		UUID savedUserID = CryptoSession.execute(Optional.of(tempPassword), null, () -> tempEncryptionContextExecutor
+				.addUserCreatedOnBuddyRequestFlush(buddyUserResource).getID());
+		return getEntityByID(savedUserID);
+	}
+
 	@Transactional
 	public UserDTO updateUser(UUID id, UserDTO userResource)
 	{
 		User originalUserEntity = getEntityByID(id);
-		UserDTO savedUser;
 		if (originalUserEntity.isCreatedOnBuddyRequest())
 		{
-			// TODO: the password needs to come from the URL, not from the
-			// payload
-			savedUser = handleUserSignupUponBuddyRequest("TODO", userResource, originalUserEntity);
-		}
-		else
-		{
-			savedUser = handleUserUpdate(userResource, originalUserEntity);
+			throw new YonaException("User is created on buddy request, use other method");
```

Comment:
Throw ``IllegalArgumentException``, as this is a programming error, and change the text into something like this: "Cannot update 'user created on buddy request' through this method".